### ISSUE TITLE
plugin_dir does not have a default value

### ISSUE THREAD

**Issue by williamtrelawny** [Graylog2/graylog2-server#15746]
## What?

If `plugin_dir` is not set or omitted in `server.conf` there is no default value it falls back to upon server start, nor does there seem to be any sort of sanity check for it having a null/empty value.

The `bin_dir` config key has a default value of `/data` and if this is omitted it is apparent in the error log that Graylog is trying to load files from the `/data` directory. While ideally I'd like both values to have sanity checks for being null/empty on startup, I would settle for just having a default value for `plugin_dir` which I assume would produce a similar type of error, leading the Graylog admin in the right direction.

## Why?

Omitting `plugin_dir` produces a huge string of cryptic Java guice errors that don't lead the troubleshooter in any direction towrad the root cause, which is actually extremely simple. Having a sane default value (like /use/share/graylog-server/plugin which it's already set to in `server.conf`), even if not the value that points Graylog to the proper plugins it needs, will likely work 90% of the time. Having no default value set at all to fallback to will always work 0% of the time and also result in completely useless error log output.

## Your Environment

* Graylog Version: 5.1

**Comment by bernd** [comment]
@williamtrelawny 

> If plugin_dir is not set or omitted in server.conf there is no default value it falls back to upon server start, nor does there seem to be any sort of sanity check for it having a null/empty value.

That's not true. If `plugin_dir` is not set in the server config file or the environment, the default value is set to `plugin`. This works for development and manual tarball installations. It has been the default for a long time.

We set the `plugin_dir` value in the default server config file according to the platform we package Graylog for. (DEB, RPM, container image)

> Omitting `plugin_dir` produces a huge string of cryptic Java guice errors that don't lead the troubleshooter in any direction towrad the root cause, which is actually extremely simple. Having a sane default value (like /use/share/graylog-server/plugin which it's already set to in `server.conf`), even if not the value that points Graylog to the proper plugins it needs, will likely work 90% of the time. Having no default value set at all to fallback to will always work 0% of the time and also result in completely useless error log output.

If the `plugin_dir` value points to a non-existent directory, the server will log a warning.

```
2023-06-30 17:25:38,765 WARN : org.graylog2.shared.plugins.PluginLoader - Plugin directory /home/bernd/graylog/test/5.1/graylog-enterprise-5.1.2-linux-x64/a/missing/plugin/dir does not exist, not loading plugins.
```

Yes, there will be lots of output because the server continues to start but is missing essential plugins. But there is an indication of it in the logs. We can probably stop the server early if the configured plugin directory cannot be found. We don't do that at the moment because, at some point, the server didn't require any plugins to run.

> The `bin_dir` config key has a default value of `/data` and if this is omitted it is apparent in the error log that Graylog is trying to load files from the `/data` directory. While ideally I'd like both values to have sanity checks for being null/empty on startup, I would settle for just having a default value for `plugin_dir` which I assume would produce a similar type of error, leading the Graylog admin in the right direction.

The default value for `bin_dir` is `bin` and not `/data`. For operating system packages and container images, the `bin_dir` value is set accordingly. I don't know an environment where we set it to `/data` by default. That seems to be a user error.

Without giving us any details about your environment, which exact version you are running, and which deployment method you are using, it's hard to help you with this, though.
